scripts/index.js

```diff
@@ -33,13 +33,16 @@
   var llliveOptGroup = document.querySelector('[label="low latency live"]');
   var manifestOptGroup = document.querySelector('[label="json manifest object"]');
 
-  // get the sources list squared away
-  var xhr = new window.XMLHttpRequest();
+  var sourceList;
+  var hlsDataManifest;
+  var dashDataManifest;
 
-  xhr.addEventListener('load', function() {
-    var sources = JSON.parse(xhr.responseText);
+  var addSourcesToDom = function() {
+    if (!sourceList || !hlsDataManifest || !dashDataManifest) {
```

Comment:
I see, this is the key to making it work. Wait till all 3 items have loaded before we process this.